readthedocs/rtd_tests/tests/test_config_integration.py

```diff
@@ -191,31 +216,67 @@ def test_extra_requirements(self, load_config):
         config = load_yaml_config(self.version)
         self.assertEqual(config.extra_requirements, [])
 
-    def test_conda(self, load_config):
-        to_find = '__init__.py'
-        load_config.side_effect = create_load({
-            'conda': {
-                'file': to_find
-            }
-        })
+    @mock.patch('readthedocs.projects.models.Project.checkout_path')
+    def test_conda_with_cofig(self, checkout_path):
+        base_path = tempfile.mkdtemp()
+        checkout_path.return_value = base_path
+        conda_file = 'environmemt.yml'
+        full_conda_file = path.join(base_path, conda_file)
+        with open(full_conda_file, 'w') as f:
+            f.write('conda')
+        create_config_file(
+            {
+                'conda': {
+                    'file': conda_file,
+                }
+            },
+            base_path=base_path,
+        )
         config = load_yaml_config(self.version)
         self.assertEqual(config.use_conda, True)
-        self.assertTrue(config.conda_file[-len(to_find):] == to_find)
+        self.assertEqual(config.conda_file, full_conda_file)
 
-        load_config.side_effect = create_load()
+    @mock.patch('readthedocs.projects.models.Project.checkout_path')
+    def test_conda_without_cofig(self, checkout_path):
+        base_path = tempfile.mkdtemp()
+        checkout_path.return_value = base_path
         config = load_yaml_config(self.version)
         self.assertEqual(config.use_conda, False)
         self.assertEqual(config.conda_file, None)
 
-    def test_requirements_file(self, load_config):
-        requirements_file = '__init__.py'
-        load_config.side_effect = create_load({
-            'requirements_file': requirements_file
-        })
+    @mock.patch('readthedocs.projects.models.Project.checkout_path')
+    def test_requirements_file_from_project_setting(self, checkout_path):
+        base_path = tempfile.mkdtemp()
+        checkout_path.return_value = base_path
+
+        requirements_file = 'requirements.txt'
+        self.project.requirements_file = requirements_file
+        self.project.save()
+
+        full_requirements_file = path.join(base_path, requirements_file)
+        with open(full_requirements_file, 'w') as f:
+            f.write('pip')
```

Comment:
Do we need to create this file at all for the test?

Reply:
Yeah, the validation is very strict, and it checks that the file exists, I wasn't able to mock this one. But I'm using pytest for our new tests, and we don't need to use this hacks anymore